src/lsptoolshost/nestedCodeAction.ts

```diff
@@ -0,0 +1,122 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as RoslynProtocol from './roslynProtocol';
+import { CodeAction, CodeActionResolveRequest, LSPAny } from 'vscode-languageserver-protocol';
+import { RoslynLanguageServer } from './roslynLanguageServer';
+import { URIConverter, createConverter } from 'vscode-languageclient/lib/common/protocolConverter';
+import { UriConverter } from './uriConverter';
+
+export function registerNestedCodeActionCommands(
+    context: vscode.ExtensionContext,
+    languageServer: RoslynLanguageServer,
+    outputChannel: vscode.OutputChannel
+) {
+    context.subscriptions.push(
+        vscode.commands.registerCommand(
+            'roslyn.client.nestedCodeAction',
+            async (request): Promise<void> => registerNestedResolveCodeAction(languageServer, request, outputChannel)
+        )
+    );
+}
+
+async function registerNestedResolveCodeAction(
+    languageServer: RoslynLanguageServer,
+    codeActionData: any,
+    outputChannel: vscode.OutputChannel
+) {
+    if (codeActionData) {
+        const data = <LSPAny>codeActionData;
+        const action = data.NestedCodeAction;
+
+        if (action && action.nestedActions && action.nestedActions.length > 0) {
+            await vscode.window
+                .showQuickPick(
+                    action.nestedActions?.map((child: { title: string }) => child.title),
+                    {
+                        placeHolder: vscode.l10n.t('Pick a nested action'),
+                        ignoreFocusOut: true,
+                    }
+                )
+                .then(async (selectedValue) => {
+                    if (selectedValue) {
+                        const selectedAction = action.nestedActions?.find(
+                            (child: { title: string }) => child.title === selectedValue
+                        );
+                        if (selectedAction) {
+                            await vscode.window.withProgress(
+                                {
+                                    location: vscode.ProgressLocation.Notification,
+                                    title: vscode.l10n.t('Nested Code Action'),
+                                    cancellable: true,
+                                },
+                                async (_, token) => {
+                                    let result: string | undefined;
+                                    if (selectedAction.data.FixAllFlavors) {
```

Comment:
Instead of having all this branching here based on flavor, could you instead extract the code from https://github.com/dotnet/vscode-csharp/blob/main/src/lsptoolshost/fixAllCodeAction.ts#L33 into an exported method that we just call from here if they picked an action with fix all?

And then this file only handles the nested application.